build/android/src/net/minetest/minetest/MinetestAssetCopy.java

```diff
@@ -95,70 +97,75 @@ protected String doInBackground(String... files)
 			m_filenames    = new Vector<String>();
 			m_tocopy       = new Vector<String>();
 			m_asset_size_unknown = new Vector<String>();
-			String baseDir = 
-					Environment.getExternalStorageDirectory().getAbsolutePath()
-					+ "/";
-			
-			
+
+			Context con = getApplicationContext();
+			String baseDir = con.getFilesDir().getAbsolutePath() + "/";
+//			String baseDir =
+//					Environment.getExternalStorageDirectory().getAbsolutePath()
+//					+ "/";
+
+
 			// prepare temp folder
-			File TempFolder = new File(baseDir + "Minetest/tmp/");
-			
+			File TempFolder = new File(baseDir
+				+ MtNativeActivity.ProjectName + "/tmp/");
```

Comment:
This is platform code already. It can be written using the File API (which exists on "normal" java too, but this never gets executed in "normal" java), the question is whether I have to change it when I only touch paths.